### PR TITLE
Stop using Rx Starter plugin

### PR DESCRIPTION
**Pull request by jimmyoak**

### CODE REVIEW

File: reactive/src/main/java/com/scmspain/howtospring/reactive/Application.java [hunk 1]
```diff
@@ -6,13 +6,12 @@
...
-import org.springframework.cloud.netflix.rx.RxJavaAutoConfiguration;
 import org.springframework.context.annotation.Import;
 import org.springframework.scheduling.annotation.EnableAsync;
 
-@EnableEurekaClient()
-@SpringBootApplication(exclude = RxJavaAutoConfiguration.class)
-@SpringCloudApplication()
+@EnableEurekaClient
+@SpringBootApplication
+@SpringCloudApplication
```
Comment (victuxbb) [inline reactive/src/main/java/com/scmspain/howtospring/reactive/Application.java]: The SpringCloudApplication annotation includes the SpringBootApplication already

Comment (jimmyoak) [inline reactive/src/main/java/com/scmspain/howtospring/reactive/Application.java]: Actually, we could remove @EnableEurekaClient too, @EnableDiscoveryClient is included by SpringCloudApplication too.

In ms-vibbo--user-rating, we only use: @SpringCloudApplication. Even @EnableAsync I think is not necessary.

Despite it's not the target of this PR, I'll remove these lines. I only wanted to remove RxJavaAutoConfig exclude statement.

File: reactive/src/main/java/com/scmspain/howtospring/reactive/Endpoint.java [hunk 2]
```diff
@@ -17,12 +22,17 @@ public Endpoint(MyService myService) {
...
     }
 
     @RequestMapping(path = "/users/{userId}", method = RequestMethod.GET)
-    public Observable<User> showUser(@PathVariable String userId) {
-        return myService.showUser(userId);
+    public Single<User> showUser(@PathVariable String userId) {
+        return myService.showUser(userId).toSingle();
+    }
+
+    @RequestMapping(path = "/users.stream", method = RequestMethod.GET)
```
Comment (jimmyoak) [inline reactive/src/main/java/com/scmspain/howtospring/reactive/Endpoint.java]: I know it's not a correct endpoint, but I didn't wanted to mess up <code>listUsers</code> action

File: reactive/src/main/resources/application.yml [hunk 3]
```diff
@@ -2,8 +2,8 @@ eureka:
   instance:
     preferIpAddress: true
   client:
-      serviceUrl:
-        defaultZone: http://192.168.99.100:8080/eureka/v2/
+    serviceUrl:
+      defaultZone: http://192.168.99.100:8080/eureka/v2/
```
Comment (jimmyoak) [inline reactive/src/main/resources/application.yml]: Indenting.

File: reactive/src/test/java/com/scmspain/howtospring/reactive/EndpointTest.java [hunk 4]
```diff
@@ -26,11 +25,7 @@ public void shouldListUsers() throws Exception {
     Endpoint controller = new Endpoint(myService);
 
     MockMvc mockServer = standaloneSetup(controller)
-        .setCustomReturnValueHandlers(
-            new ObservableReturnValueHandler(),
-            new SingleReturnValueHandler(),
-            new org.springframework.cloud.netflix.rx.SingleReturnValueHandler()
-        ).build();
+        .setCustomReturnValueHandlers(new SingleReturnValueHandler()).build();
```
Comment (jimmyoak) [inline reactive/src/test/java/com/scmspain/howtospring/reactive/EndpointTest.java]: Removed Rx plugin value handlers.

**Comment by jimmyoak** [comment]
@aramirez-es @fiunchinho @victuxbb @Arkioner

**Comment by jimmyoak** [comment]
@kmruiz

**Comment by fiunchinho** [comment]
THANK YOU so much for the PR.
So the @Async annotation is no longer needed? I think is needed if we want Tomcat NIO

If you could just squash the commits, you'd earn all my respects.

**Comment by jimmyoak** [comment]
Squashed.

What is the point of @EnableAsync here? @EnableAsync should be used with @Async annotation, I'm not using it anywhere in ms-vibbo-user-rating.

**Comment by fiunchinho** [comment]
:+1:

**Pull request merged**